### ISSUE TITLE
Allow textarea key presses to be synchronous?

### ISSUE THREAD

**Issue by rntz** [talonvoice/talon#259]
**Problem**: synchronizing between key presses and access to a text area's attributes/content. For example, "go word left" (which presses ctrl-left) followed immediately by accessing the cursor location and content of the text area to determine how to insert some text in a context-sensitive way.

**Solution?**: Some way to make key presses to talon-controlled textareas synchronous instead of asynchronous.

_Copying from slack:_

ok, so, technically the textarea handles keystrokes entirely manually

so if I translate qt key/modifier names to talon key names, then you could actually just fire actions.key right into the textarea

maybe file an issue asking for that, that's probably the easiest way for you to not think about sync

something like this:

```
override actions.key(key string):
    if not textarea.key(key string):
        actions.next(key string)
```

basically offer all actions.key()s to the textarea first, and any that it doesn't know how to handle just send to the system instead

that needs
1. qt key names need to be translated to talon ones in Canvas
2. a public textarea method that can perform key simulation

**Comment by timo** [comment]
why don't we literally have a method on the textarea widget that moves the cursor left by one word then we don't have to go through the whole key / event / whatever dance at all?

**Comment by rntz** [comment]
That would solve this particular use case, if we wired the right action up, but it means if even a single action doesn't get wired up to a special-case textarea implementation and instead presses some keys, you have a potential synchronization bug.

Mind you, I'm not saying it's a *bad* idea to make special case implementations of the editing actions for textarea, it's probably a good idea. Not sure whether that requires any more support from talon core, we can probably do a lot of it already.
